apps/admin-gui/src/app/shared/components/dialogs/update-application-form-dialog/update-application-form-dialog.component.ts

```diff
@@ -34,21 +34,45 @@ export class UpdateApplicationFormDialogComponent implements OnInit {
   ngOnInit(): void {
     this.theme = this.data.theme;
     this.applicationForm = this.data.applicationForm;
-    this.moduleName = this.applicationForm.moduleClassName;
+    this.moduleNames = this.applicationForm.moduleClassNames;
+    if (!this.moduleNames) {
+      this.moduleNames = [''];
+    } else if (this.moduleNames.length === 0) {
+      this.moduleNames.push('');
+    }
     this.initialState = this.applicationForm.automaticApproval ? 'auto' : 'manual';
     this.extensionState = this.applicationForm.automaticApprovalExtension ? 'auto' : 'manual';
     this.embeddedState = this.applicationForm.automaticApprovalEmbedded ? 'auto' : 'manual';
     this.entity = this.data.entity;
     this.autoRegistrationEnabled = this.data.autoRegistrationEnabled;
   }
 
+  addModule(): void {
+    this.moduleNames.push('');
+  }
+
+  removeModule(index: number): void {
+    this.moduleNames.splice(index, 1);
+  }
+
+  trackByIndex(index: number): number {
+    return index;
+  }
+
+  preventComma(event: KeyboardEvent): void {
+    if (event.key === ',') {
+      event.preventDefault();
+    }
+  }
```

Comment:
Would prefer if we showed error on the mat form that says commas are not allowed.